File: creational/singleton/simple-example/python/singleton.py
class Singleton:
    __instance = None

    def __new__(cls, *args):
        if cls.__instance is None:
            cls.__instance = object.__new__(cls, *args)
        return cls.__instance

# python3 singleton.py
if __name__ == "__main__":
    # The client code.
    s1 = Singleton()
    s2 = Singleton()

    if id(s1) == id(s2):
        print("Singleton works, both variables contain the same instance.")
    else:
        print("Singleton failed, variables contain different instances.")

File: creational/singleton/simple-example/golang/singleton.go
package main

import "fmt"

// Note this isn't concurrent safe!
type Singleton struct {
	// ... other fields here
}

var instance *Singleton

func GetInstance() *Singleton {
	if instance != nil {
		return instance
	}

	instance = new(Singleton)
	return instance
}

// GO111MODULE="off" go run singleton.go
func main() {
	s1 := GetInstance()
	s2 := GetInstance()
	if s1 == s2 {
		fmt.Println("Singleton works, both variables contain the same instance.")
	} else {
		panic("Singleton instances should be the same")
	}
}
